packages/state-transition/src/cache/epochCache.ts

```diff
@@ -549,6 +571,20 @@ export class EpochCache {
     // ```
     this.epoch = computeEpochAtSlot(state.slot);
     this.syncPeriod = computeSyncPeriodAtEpoch(this.epoch);
+
+    // To populate finalized cache and prune unfinalized cache with validators that just entered the activation queue
+    if (this.isAfterEIP6110()) {
+      const expectedActivationEligibilityEpoch = prevEpoch + 1;
```

Comment:
Why do prevEpoch + 1 instead of using currEpoch? If it's to mimic some code elsewhere, reference it